### PR TITLE
update to data driven - superseeds part of https://github.com/qmk/qmk…

### PR DESCRIPTION
**Pull request by alinelena**
…_firmware/pull/18468

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by alinelena** [comment]
this was reviewed in #18468

File: keyboards/mlego/m65/rev1/info.json [hunk 1]
```diff
@@ -1,18 +1,136 @@
...
+            "twinkle": true
+        },
+        "hue_steps": 8,
+        "layers": {
+            "enabled": true
+        },
+        "led_count": 20,
+        "pin": "B15",
+        "saturation_steps": 8
+    },
```
Comment (drashna) [inline keyboards/mlego/m65/rev1/info.json:58]:
```suggestion
        "saturation_steps": 8
    },
    "ws2812": {
        "pin": "B15"
    },
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m65/m65.c [hunk 2]
```diff
@@ -191,14 +174,90 @@ oled_rotation_t oled_init_user(oled_rotation_t rotation) {
...
+    return false;
+}
+
+void matrix_scan_user(void) {
+
+    toggle_leds();
+
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t* record) {
```
[the remark this answers is not in the document]
Comment (alinelena) [reply keyboards/mlego/m65/m65.c:207]: done

File: keyboards/mlego/m65/m65.c [hunk 3]
```diff
@@ -191,14 +174,90 @@ oled_rotation_t oled_init_user(oled_rotation_t rotation) {
...
+                set_led_toggle(_RSE, !layer_state_is(_RSE));
+            }
+            return true;
+            break;
+        default:
+            return true;
+    }
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
```
[the remark this answers is not in the document]
Comment (alinelena) [reply keyboards/mlego/m65/m65.c]: done

File: keyboards/mlego/m65/keymaps/default/keymap.c [hunk 4]
```diff
@@ -51,69 +37,11 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
-    rgblight_layers = my_rgb();
-
-#endif
-
-#ifdef OLED_ENABLE
-
-    init_timer();
-
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (fauxpark) [inline keyboards/mlego/m65/keymaps/default/keymap.c:41]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/mlego/m65/keymaps/uk/keymap.c [hunk 5]
```diff
@@ -1,105 +1,108 @@
...
+    [rv]   = 0x2285,  // ⊅
+    [rV]   = 0x2287,  // ⊇
+    [rb]   = 0x2227,  // ∧
+    [rB]   = 0x22BC,  // ⊼
+    [rn]   = 0x2228,  // ∨
+    [rN]   = 0x22BB,  // ⊻
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (fauxpark) [inline keyboards/mlego/m65/keymaps/uk/keymap.c:99]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m65/m65.c [hunk 6]
```diff
@@ -191,14 +175,90 @@ oled_rotation_t oled_init_user(oled_rotation_t rotation) {
...
+        clear_screen();
+        user_oled_magic();
     }
     return false;
 }
 
 #endif
+
+
+bool led_update_kb(led_t led_state) {
```
Comment (drashna) [inline keyboards/mlego/m65/m65.c:193]: All of these functions need to call the user version, somewhere.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please run `qmk format-json` on all info.json files.

**Comment by alinelena** [comment]
> Please run `qmk format-json` on all info.json files.

done, thank you!

File: keyboards/mlego/m65/rev2/info.json [hunk 7]
```diff
@@ -1,42 +1,134 @@
...
-    },
-    "matrix_pins": {
-        "cols": ["A10", "A15", "B3", "B4", "B5", "B9", "B8", "B7", "B6", "C15", "A0", "A7", "B10"],
-        "rows": ["B11", "B0", "B1", "A2", "A3"]
-    },
+    "manufacturer": "Alin Elena",
+    "keyboard_name": "mlego/m65 rev2",
+    "maintainer": "alin elena",
+    "bootloader": "stm32duino",
+    "debounce": 5,
```
Comment (drashna) [inline keyboards/mlego/m65/rev2/info.json:6]:
```suggestion
```

File: keyboards/mlego/m65/rev3/info.json [hunk 8]
```diff
@@ -1,43 +1,135 @@
...
-    "matrix_pins": {
-        "cols": ["A10", "A15", "B3", "B4", "B5", "B9", "B8", "B7", "A1", "A2", "A3", "A4", "B1"],
-        "rows": ["B10", "A5", "A6", "A7", "B0"]
-    },
+    "manufacturer": "Alin Elena",
+    "keyboard_name": "mlego/m65 rev3",
+    "maintainer": "alin elena",
+    "board": "BLACKPILL_STM32_F401",
+    "bootloader": "tinyuf2",
+    "debounce": 5,
```
Comment (drashna) [inline keyboards/mlego/m65/rev3/info.json:7]:
```suggestion
```

File: keyboards/mlego/m65/rev4/info.json [hunk 9]
```diff
@@ -1,43 +1,137 @@
...
-    "matrix_pins": {
-        "cols": ["B14", "A8", "A10", "A15", "B3", "B4", "B5", "B7", "A1", "A2", "A3", "A4", "B1"],
-        "rows": ["B10", "A5", "A6", "A7", "B0"]
-    },
+    "manufacturer": "Alin Elena",
+    "keyboard_name": "mlego/m65 rev4",
+    "maintainer": "alin elena",
+    "board": "BLACKPILL_STM32_F401",
+    "bootloader": "tinyuf2",
+    "debounce": 5,
```
Comment (drashna) [inline keyboards/mlego/m65/rev4/info.json:7]:
```suggestion
```

File: keyboards/mlego/m65/rev4/rules.mk [hunk 10]
```diff
@@ -1,15 +1 @@
...
-CONSOLE_ENABLE = no        # Console for debug
-COMMAND_ENABLE = no        # Commands for debug and configuration
-NKRO_ENABLE = yes          # Enable N-Key Rollover
-BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = yes      # Enable keyboard RGB underglow
-AUDIO_ENABLE = no          # Audio output
-ENCODER_ENABLE = yes       # Enable encoder
-OLED_ENABLE = yes          # Enable OLED
-WPM_ENABLE = yes           # Enable WPM Counter 
+OLED_DRIVER = SSD1306      # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
```
Comment (drashna) [inline keyboards/mlego/m65/rev4/rules.mk:1]: This is the default, and should be removed.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — COMMENTED** [review]
CI is reporting an unterminated ifdef, but it's hard to see it in all the changes here.

**Comment by alinelena** [comment]
> CI is reporting an unterminated ifdef, but it's hard to see it in all the changes here.
> 
> Ah, introduced here: [63063a4](https://github.com/qmk/qmk_firmware/commit/63063a494308358607f22f8cc4d4c210ef79debd)

done thank you! too eager on dd

**Comment by alinelena** [comment]
> Please run `qmk format-json` on all info.json files.

done!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**